internal/metrics/metrics.go

```diff
@@ -0,0 +1,214 @@
+	// Labels: namespace, deployment_name, action (scale_up/scale_down)
+	ActualScalingCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_scaling_actions_total",
+			Help: "Total number of actual scaling actions performed by the controller",
+		},
+		[]string{"namespace", "deployment_name", "action"},
+	)
+
+	// PDBCreationCounter tracks PDB creation events
+	// Labels: namespace, deployment_name
+	PDBCreationCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_pdb_creations_total",
+			Help: "Total number of PDBs created by the eviction autoscaler",
+		},
+		[]string{"namespace", "deployment_name"},
+	)
+
+	// EvictionAutoScalerCreationCounter tracks EvictionAutoScaler creation events
+	// Labels: namespace, pdb_name, target_deployment
+	EvictionAutoScalerCreationCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_creation_total",
+			Help: "Total number of EvictionAutoScaler resources created",
+		},
+		[]string{"namespace", "pdb_name", "target_deployment"},
+	)
+
+	// NodeCordoningCounter tracks node cordoning events detected
+	// Labels: node_name, cordoned (true/false)
+	NodeCordoningCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_node_cordoning_total",
+			Help: "Total number of node cordoning events detected by the eviction autoscaler",
+		},
+		[]string{"node_name", "cordoned"},
+	)
+
+	// NodeDrainCounter tracks node drain events detected (pods evicted from cordoned nodes)
+	// Labels: node_name, pod_name, namespace
+	NodeDrainCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_node_drain_total",
+			Help: "Total number of pods evicted from nodes during drain operations",
+		},
+		[]string{"node_name", "pod_name", "namespace"},
+	)
+)
+
+func init() {
+	// Register metrics with controller-runtime's registry
+	// Use safe registration to prevent panics
+	collectors := []prometheus.Collector{
+		DeploymentGauge,
+		PDBGauge,
+		EvictionCounter,
+		BlockedEvictionCounter,
+		ScalingOpportunityCounter,
+		ActualScalingCounter,
+		PDBCreationCounter,
+		EvictionAutoScalerCreationCounter,
+		NodeCordoningCounter,
+		NodeDrainCounter,
+	}
+
+	for _, collector := range collectors {
+		if err := metrics.Registry.Register(collector); err != nil {
+			// Log error but don't crash - metrics are not critical for controller function
+			// In production, you might want to use a proper logger here
+			// For now, we'll just continue without that specific metric
+			continue
+		}
+	}
+}
+
+// Helper functions to update metrics
+
+// UpdateDeploymentCount updates the deployment gauge
+func UpdateDeploymentCount(namespace string, canCreatePDB bool, count float64) {
+	canCreatePDBStr := "false"
+	if canCreatePDB {
+		canCreatePDBStr = "true"
+	}
+	DeploymentGauge.WithLabelValues(namespace, canCreatePDBStr).Set(count)
```

Comment:
We probably want an incremennt and decrement rather than set because of the count